text/0000-capture-disjoint-fields.md

```diff
@@ -0,0 +1,196 @@
+- Feature Name: capture-disjoint-fields
+- Start Date: 2017-11-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes that closure capturing should be minimal rather than maximal.
+Specifically, existing rules regarding borrowing and moving disjoint fields
+should be applied to capturing. If implemented, the following code examples
+would become valid:
+
+```rust
+let _a = &mut foo.a;
+|| &mut foo.b; // Error! cannot borrow `foo`
+```
+
+```rust
+let _a = &mut foo.a;
+move || foo.b; // Error! cannot move `foo`
+```
+
+Note that there is open issue relating to this RFC
+([#19004](https://github.com/rust-lang/rust/issues/19004)) where some discussion
+has already taken place.
+
+# Motivation
+[motivation]: #motivation
+
+In the rust language today, any variables named within a closure will be fully
+captured. This was simple to implement but is inconstant with the rest of the
+language because rust normally allows simultaneous borrowing of disjoint fields.
+Remembering this exception adds to the mental burden of the programmer and
+worsens rust's already terrible learning curve.
```

Comment:
Terrible sounds a bit strong - I'd agree to Rust having a steep learning curve, but terrible is a bit much.